src/build.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const core = require('@actions/core')
+const { exec } = require('@actions/exec')
+const parseReleaseMetadata = require('./utils/parseReleaseMetadata')
+
+function getMonorepoData({ context, inputs, github }) {
+  const pr = context?.payload?.pull_request
+  const isOpticPr = pr?.user.login === inputs['app-name']
+
+  if (github.event_name === 'pull_request' && isOpticPr) {
+    return parseReleaseMetadata(pr)
+  }
+
+  return {
+    monorepoPackage: inputs['monorepo-package'],
+    monorepoRoot: inputs['monorepo-root'],
+  }
+}
+
+module.exports = async function ({ github, context, inputs }) {
+  try {
+    const { monorepoPackage, monorepoRoot } = getMonorepoData({
+      context,
+      inputs,
+      github,
+    })
+
+    const buildCommands = inputs['build-command']
+      .trim()
+      .split('\n')
+      .filter(cmd => !!cmd)
+      .map(cmd =>
+        cmd
+          .trim()
+          .split(' ')
+          .filter(c => !!c)
+      )
+
+    const options = {
+      cwd: monorepoPackage ? `${monorepoRoot}/${monorepoPackage}` : '.',
```

Comment:
as far as I can see
- monorepoRoot is only used for generating directory names
- monorepoPackage could contain a slash

...which kind of makes monorepo-root redundant in its current form? (other than my point above about supporting monorepos in a sub folder)